### PR TITLE
Feat: Cart , Wishlist, Alert 

### PR DESCRIPTION
**Pull request by Logan1x**
This PR includes changes in 
- Added cart feature
- Added Wishlist Feature
- Added Alert Feat
- Added Move to cart/wishlist from product
- Added Move to cart from wishlist
- Added 404

Deploy Preview: https://deploy-preview-3--plethora-v2.netlify.app/

### CODE REVIEW

File: src/components/nav.js [hunk 1]
```diff
@@ -64,8 +59,19 @@ export default function Nav() {
...
             <path d="M3 1a1 1 0 000 2h1.22l.305 1.222a.997.997 0 00.01.042l1.358 5.43-.893.892C3.74 11.846 4.632 14 6.414 14H15a1 1 0 000-2H6.414l1-1H14a1 1 0 00.894-.553l3-6A1 1 0 0017 3H6.28l-.31-1.243A1 1 0 005 1H3zM16 16.5a1.5 1.5 0 11-3 0 1.5 1.5 0 013 0zM6.5 18a1.5 1.5 0 100-3 1.5 1.5 0 000 3z"></path>
           </svg>
-          <span className="badge">2</span>
-        </a>
+          {state.cartData.length > 0 && (
+            <span className="badge">{state.cartData.length}</span>
+          )}
+        </Link>
+        {token ? (
+          <a onClick={logoutHandler} className=" btn-filled">
```
Comment (parthpandyappp) [inline src/components/nav.js]: Remove unwanted spaces from `className`

File: src/contexts/dataContext.js [hunk 2]
```diff
@@ -1,30 +1,101 @@
...
-      if (response.status === 200 || response.status === 201) {
-        return response.data;
-      }
-    } catch (error) {
-      console.log(error);
-    }
-  };
+  const { token } = useAuth();
+
+  // const getData = async () => {
```
Comment (parthpandyappp) [inline src/contexts/dataContext.js]: Remove redundant comments

**Review by parthpandyappp — APPROVED** [review]
I've left some comments. KIndly look into it and then move ahead with merge.

**Review by shrey27 — APPROVED** [review]
UI is working as expected
Code structure looks fine
PR is good to merge

File: src/contexts/authContext.js [hunk 3]
```diff
@@ -1,6 +1,8 @@
+import { useNavigate } from "react-router-dom";
 import { createContext, useContext, useState } from "react";
+
 import { loginUtility } from "../utils/utilAuthFuncs";
-import { useNavigate } from "react-router-dom";
+import { notify } from "../utils/utilToastFuncs";
```
Comment (SJTGSHIVAM) [inline src/contexts/authContext.js:6]: Just a suggestion you can configure you config files to have absolute imports it is really a life saver, must try it out

File: src/contexts/authContext.js [hunk 4]
```diff
@@ -23,25 +25,28 @@ const AuthProvider = ({ children }) => {
...
         );
-        navigate("/");
+        navigate("/products");
         setToken(response.data.encodedToken);
         setCurrUser(response.data.foundUser);
+        notify("Logged in successfully!", "success");
       }
     } catch (err) {
-      console.log(err);
+      console.error(err);
```
Comment (SJTGSHIVAM) [inline src/contexts/authContext.js:34]: refrain from printing errors to console. users must know but they must not get into debugging our code

File: src/hooks/filterHook.js [hunk 5]
```diff
@@ -4,7 +4,10 @@ import {
   filterByPriceRange,
 } from "../utils/utilFilterFuncs";
 
+// import { updateCartItemQuantity } from "../utils/utilCartFuncs";
+
 import { useDataContext } from "../contexts/dataContext";
+// import { updateCartItemQuantity } from "../utils/utilCartWishFuncs";
 
 export const useFilterHook = () => {
```
Comment (shrey27) [inline src/hooks/filterHook.js:12]: this approach is very handy for handling filters

File: src/contexts/dataContext.js [hunk 6]
```diff
@@ -1,30 +1,119 @@
...
+  getData,
+  getCartData,
+  getWishlistData,
+  postCartData,
+  postWishlistData,
+} from "../utils/utilDataFuncs";
+
+import {
+  removeFromCartUtilFunc,
+  removeFromWishlistUtilFunc,
```
Comment (SJTGSHIVAM) [inline src/contexts/dataContext.js:14]: large names are good but remember we should not exceed a certain value (26-32) and can pick any threshold as per your choice. 
No problem is there in your code. I mentioned it just for discussing

**Review by SJTGSHIVAM — APPROVED** [review]
The code looks good I have added some discussion points please check them out. We are good to go for now.

File: src/contexts/dataContext.js (hunk 6 above)
Comment (Logan1x) [reply src/contexts/dataContext.js:14]: Thanks will keep them in mind next time :)

File: src/contexts/authContext.js (hunk 4 above)
Comment (Logan1x) [reply src/contexts/authContext.js:34]: Good point. will add toast in the upcoming version.

File: src/contexts/authContext.js (hunk 3 above)
Comment (Logan1x) [reply src/contexts/authContext.js:6]: Could not understand this one, Can you give a blog/video link?

**Comment by Logan1x** [comment]
Thanks everyone, merging now :)

**Pull request merged**